src/python/pants/backend/python/lint/isort/subsystem.py

```diff
@@ -2,9 +2,18 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.option.custom_types import file_option
 
 
 class Isort(PythonToolBase):
   options_scope = 'isort'
   default_requirements = ['isort==4.3.20', 'setuptools']
   default_entry_point = 'isort.main'
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register(
+      '--config', type=file_option, default=None,
+      help="Path to `isort.cfg` or alternative isort config file"
+    )
```

Comment:
This won't do anything in V1, but it doesn't need to because isort will still automatically find their config in their project. The reason we need this in V2 is to ensure that the config file is included in the inputted snapshot to the ExecuteProcessRequest.

Reply:
There's no way to pass it to the V1 task. isort doesn't allow specifying the config file :/

Reply:
wat

Reply:
Another fun detail about isort is that it will leak your root level config file, if you have one. There are also over 4 different accepted config formats. The author of the project said he regretted giving so many options when weighing in on Black deciding if they wanted to support anything other than `pyproject.toml` for their config.